File: services/ai/app/services/ffmpeg_service.py
import os
import subprocess

def extract_audio_from_media(media_path: str, output_dir: str) -> str:
    base, _ = os.path.splitext(os.path.basename(media_path))
    output_audio = os.path.join(output_dir, base + '.wav')
    cmd = [
        'ffmpeg', '-i', media_path, '-vn', '-acodec', 'pcm_s16le', '-ar', '16000', '-ac', '1', output_audio
    ]
    try:
        subprocess.run(cmd, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        return output_audio
    except subprocess.CalledProcessError as e:
        raise RuntimeError(f'ffmpeg failed: {e.stderr.decode()}')

File: backend/src/services/ffmpegService.ts
import { spawn } from 'child_process';
import path from 'path';

export async function extractAudioFromMedia(mediaPath: string, outputDir: string): Promise<string> {
  return new Promise((resolve, reject) => {
    const outputAudio = path.join(outputDir, path.basename(mediaPath, path.extname(mediaPath)) + '.wav');
    const ffmpeg = spawn('ffmpeg', ['-i', mediaPath, '-vn', '-acodec', 'pcm_s16le', '-ar', '16000', '-ac', '1', outputAudio]);

    ffmpeg.on('close', (code) => {
      if (code === 0) {
        resolve(outputAudio);
      } else {
        reject(new Error(`ffmpeg exited with code ${code}`));
      }
    });
    ffmpeg.on('error', (err) => reject(err));
  });
}
